components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/Utils.java

```diff
@@ -143,4 +147,93 @@ public static String getProperty(File srcFile, String key) throws IOException {
         }
         return value;
     }
+
+    /**
+     * This method will return the path to logs directory.
+     * @return  path as string.
+     */
+    public static String getCarbonLogsPath() {
+
+        String carbonLogsPath = System.getProperty("carbon.logs.path");
+        if (carbonLogsPath == null) {
+            carbonLogsPath = System.getenv("CARBON_LOGS");
+            if (carbonLogsPath == null) {
+                return getCarbonHome() + File.separator + "repository" + File.separator + "logs";
+            }
+        }
+        return carbonLogsPath;
+    }
+
+    /**
+     * This method will return the path to CARBON_HOME.
+     * @return  path as String.
+     */
+    public static String getCarbonHome() {
+
+        String carbonHome = System.getProperty("carbon.home");
+        if (carbonHome == null) {
+            carbonHome = System.getenv("CARBON_HOME");
+            System.setProperty("carbon.home", carbonHome);
+        }
+        return carbonHome;
+    }
+
+    /**
+     * This method will provide information on all the files in the logs directory.
+     * @return list of file info objects.
+     */
+    public static List<LogFileInfo> getLogFileInfoList() {
+
+        String folderPath = Utils.getCarbonLogsPath();
+        List<LogFileInfo> logFilesList = new ArrayList<>();
+        LogFileInfo logFileInfo;
+
+        File folder = new File(folderPath);
+        File[] listOfFiles = folder.listFiles();
+
+        if (listOfFiles == null) {
```

Comment:
Shouldn't we check for the size of listOfFiles == 0 as well?

Reply:
This will never be empty since there's a default response given by getDefaultLogInfoList()

Reply:
+1 Fixed